[stable/rabbitmq-ha] Add RabbitMQ Prometeus Plugin

Gsantomaggio opened the pull request:
-->
#### Is this a new chart
No
#### What this PR does / why we need it:
With rabbitmq 3.8 the rabbitmq prometheus plugin is shipped by default
With this commit is possible to enable/disable it

#### Which issue this PR fixes

#### Special notes for your reviewer:
This plugin is shipped only with RabbitMQ 3.8.
It is useful for people (like me ) that need to use RabbitMQ 3.8.
Thank you

cc @steven-sheehy 
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

Signed-off-by: Gabriele Santomaggio <g.santomaggio@gmail.com>

steven-sheehy commented on stable/rabbitmq-ha/values.yaml:
```diff
@@ -253,6 +253,15 @@ rabbitmqWebSTOMPPlugin:
 
+## Prometheus Plugin
+## Ref: https://www.rabbitmq.com/prometheus.html
+rabbitmqPrometheus:
```
Should be `rabbitmqPrometheusPlugin`

Gsantomaggio replied:
Ok, will change

steven-sheehy commented on stable/rabbitmq-ha/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 appVersion: 3.7.19
-version: 1.33.1
+version: 1.33.2
```
`1.34.0` for new features

Gsantomaggio replied:
Perfect

steven-sheehy commented on stable/rabbitmq-ha/README.md:
```diff
@@ -142,6 +142,7 @@ and their default values.
 | `rabbitmqWebMQTTPlugin.enabled`                | Enable MQTT over websocket plugin                                                                                                                                                                     | `false`                                                    |
 | `rabbitmqWebSTOMPPlugin.config`                | STOMP over websocket configuration                                                                                                                                                                    | ``                                                         |
 | `rabbitmqWebSTOMPPlugin.enabled`               | Enable STOMP over websocket plugin                                                                                                                                                                    | `false`                                                    |
+| `rabbitmqPrometheus.enabled`               | Enable native RabbitMQ prometheus plugin                                                                                                                                                                    | `false`                                                    |
```
Add a note to the prometheus section at the bottom in readme about if you're using RabbitMQ 3.8 you should not use `prometheus.exporter.enabled=true` but use the plugin instead.

Gsantomaggio replied:
makes sense :)!

steven-sheehy reviewed (changes requested):
So you've tested this chart with RabbitMQ 3.8 and it works fine with only this change necessary? If so, that's good to hear. We should probably bump the RabbitMQ version in the near future.

steven-sheehy commented on stable/rabbitmq-ha/README.md:
```diff
@@ -256,7 +257,15 @@ the following keys:
+
+To enable metrics via the traditional rabbitmq_exporter, set `prometheus.enabled` to `true`. See `values.yaml` file for more details and configuration options.
+
+While there are two ways of monitoring RabbitMQ with Prometheus, both  `prometheus.enabled` and `rabbitmqPrometheusPlugin.enabled`can be enabled at the same time, they are not mutually exclusive. 
```
Remove this line. Yes, technically they can be ran at the same time since they're different ports but my point about mutually exclusivity was that you wouldn't generally want to run both at the same time and it could cause incorrect metrics if they both update a metric with the same name.

Gsantomaggio replied:
I removed the section, but let me say that native and HTTP are two different kinds of monitoring. 
>it could cause incorrect metrics 

I'd say  "incorrect reading"  instead of "incorrect metrics ":)! 
btw I agree with you, you have to know that native is for single node and http is for whole the cluster.

steven-sheehy replied:
Not sure what you mean by native vs HTTP. Native in the sense that it's built into RabbitMQ as a plugin, I guess. But they're both HTTP endpoints that Prometheus scrapes.

gerhard replied:
Yes, native in this context means RabbitMQ plugin, which runs within the same system process as RabbitMQ, the Erlang VM.

gerhard replied:
rabbitmq_exporter gets the metrics from the RabbitMQ Management HTTP API, and then serves them from its own TCP listener.

rabbitmq-prometheus, the plugin, runs and reads metrics directly from within the Erlang VM where RabbitMQ runs, and exposes them via a dedicated TCP listener. This TCP listener is owned by the same system process that runs RabbitMQ - it's the same Erlang VM. All this is wrapped in the `native` claim.

Would you agree @brian-brazil?

brian-brazil replied:
That sounds about right to me. Using both shouldn't cause a problem, though I'd probably clarify that you usually only want to scrape one of them.

steven-sheehy commented on stable/rabbitmq-ha/README.md:
```diff
@@ -256,7 +257,15 @@ the following keys:
+
+Team RabbitMQ manages Grafana dashboards that are meant to be used with the native Prometheus support. They are publicly available at [grafana.com/orgs/rabbitmq](https://grafana.com/orgs/rabbitmq).
+
+To enable metrics via the traditional rabbitmq_exporter, set `prometheus.enabled` to `true`. See `values.yaml` file for more details and configuration options.
```
`traditional rabbitmq_exporter sidecar container`. Also make rabbitmq_exporter a link to https://github.com/kbudde/rabbitmq_exporter

Gsantomaggio replied:
fixed here: https://github.com/helm/charts/pull/17925/files#diff-a03c73b511fa43f47f906cf3fb1677d5R266

steven-sheehy commented on stable/rabbitmq-ha/values.yaml:
```diff
@@ -253,6 +253,15 @@ rabbitmqWebSTOMPPlugin:
+rabbitmqPrometheusPlugin:
+  enabled: false
+  ## Prometheus configuration:
+  config: |
```
I don't see how this config is used anywhere.

Gsantomaggio replied:
sorry you are right, fixed here https://github.com/helm/charts/pull/17925/commits/4ff4bf79367310d2f7e43ae8a997ee57fda7064b#diff-28e0ed5d78530ec2e1cf87dab1eb62a5R115-R119

steven-sheehy commented on stable/rabbitmq-ha/README.md:
```diff
@@ -256,7 +257,13 @@ the following keys:
+
+Team RabbitMQ manages Grafana dashboards that are meant to be used with the native Prometheus support. They are publicly available at [grafana.com/orgs/rabbitmq](https://grafana.com/orgs/rabbitmq).
+
+To enable metrics via the raditional [rabbitmq_exporter](https://github.com/kbudde/rabbitmq_exporter) sidecar container, set `prometheus.enabled` to `true`. See `values.yaml` file for more details and configuration options.
```
`traditional`

steven-sheehy reviewed (changes requested):
One small spelling error and it looks good

Gsantomaggio committed:
[stable/rabbitmq-ha] Add RabbitMQ Prometeus Plugin

With rabbitmq 3.8 the prometheus plugin is shipped by default [1]
With this commit is possible to enable/disable it

1-https://www.rabbitmq.com/prometheus.html

Signed-off-by: Gabriele Santomaggio <g.santomaggio@gmail.com>

steven-sheehy reviewed (approved):
/lgtm
Thanks!
